includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -517,6 +517,51 @@ public function save_payment_method_checkbox( $force_checked = false ) {
 		<?php
 	}
 
+	/**
+	 * Adds customer data to 'wcpay-checkout' when on 'Pay for Order' or 'Add Payment Method' pages.
+	 * Customer data is retrieved from order when on Pay for Order.
+	 * Customer data is retrieved from customer when on 'Add Payment Method'
+	 */
+	public function maybe_localize_customer_data() {
+		if ( ! isset( $_GET['pay_for_order'] ) && ! is_add_payment_method_page() ) {
+			return;
+		}
+
+		global $wp;
+		$user_email = '';
+		$firstname  = '';
+		$lastname   = '';
+
+		if ( isset( $_GET['pay_for_order'] ) && 'true' === $_GET['pay_for_order'] ) {
+			$order_id = wc_clean( $wp->query_vars['order-pay'] );
+			$order    = wc_get_order( $order_id );
+
+			if ( is_a( $order, 'WC_Order' ) ) {
+				$firstname  = $order->get_billing_first_name();
+				$lastname   = $order->get_billing_last_name();
+				$name       = $firstname . ' ' . $lastname;
+				$user_email = $order->get_billing_email();
+			}
+		}
+
+		if ( is_add_payment_method_page() ) {
+			$user = wp_get_current_user();
+
+			if ( $user->ID ) {
+				$firstname  = $user->user_firstname;
+				$lastname   = $user->user_lastname;
+				$user_email = get_user_meta( $user->ID, 'billing_email', true );
+				$user_email = $user_email ? $user_email : $user->user_email;
```

Comment:
```suggestion
				$user_email = get_user_meta( $user->ID, 'billing_email', true ) ?: $user->user_email;
```
I find this simpler and more clear what the code is doing.

Reply:
Yes, it looks cleaner but our PHPCS rules do not allow for short ternaries...

Reply:
That's probably a rule from PHP 5.2 days. Maybe we could change it at some point. This is fine for now.